> From my experience on slightly older tree,
a few old programs incorrectly rely on ncurses internal details
and may now require setting NCURSES_INTERNAL=1 to allow this badness.

Oh, interesting. Do you happen to know this build failure could be avoided with that? (It started happening ever since I couldn't figure how to get ncurses5 compiling on gcc7 and switched to "--with-abi-version=5" instead on the gcc-7 branch).

````
cc -g -O2  -DHAVE_CONFIG_H  -c mach_dep.c
mach_dep.c: In function 'lock_sc':
mach_dep.c:406:2: warning: ignoring return value of 'fgets', declared with attribute warn_unused_result [-Wunused-result]
  (void) fgets(prbuf, MAXSTR, stdin);
  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
gcc -g -O2  -DHAVE_CONFIG_H  -c main.c
gcc -g -O2  -DHAVE_CONFIG_H  -c mdport.c
mdport.c: In function 'md_hasclreol':
mdport.c:264:17: error: dereferencing pointer to incomplete type 'TERMINAL {aka struct term}'
     if (cur_term->type.Strings == NULL)
                 ^~
make: *** [Makefile:130: mdport.o] Error 1
````

https://hydra.nixos.org/build/69789236

--- a/pkgs/development/libraries/ncurses/default.nix
+++ b/pkgs/development/libraries/ncurses/default.nix
@@ -15,7 +15,7 @@ stdenv.mkDerivation rec {
   name = "ncurses-${version}" + lib.optionalString (abiVersion == "5") "-abi5-compat";
 
   src = fetchurl {
-    url = "mirror://gnu/ncurses/${name}.tar.gz";
+    url = "mirror://gnu/ncurses/ncurses-${version}.tar.gz";
     sha256 = "05qdmbmrrn88ii9f66rkcmcyzp1kb1ymkx7g040lfkd1nkp7w1da";
   };
 
@@ -117,6 +117,8 @@ stdenv.mkDerivation rec {
     moveToOutput "bin/tic" "$out"
     moveToOutput "bin/tput" "$out"
     moveToOutput "bin/tset" "$out"
+    moveToOutput "bin/captoinfo" "$out"
+    moveToOutput "bin/infotocap" "$out"
   '';
 
   preFixup = lib.optionalString (!hostPlatform.isCygwin) ''

--- a/pkgs/development/libraries/ncurses/default.nix
+++ b/pkgs/development/libraries/ncurses/default.nix
@@ -11,18 +11,12 @@
 }:
 
 stdenv.mkDerivation rec {
-  version = "6.0-20171125";
+  version = "6.1";
   name = "ncurses-${version}" + lib.optionalString (abiVersion == "5") "-abi5-compat";
 
   src = fetchurl {
-    urls = [
-      # Remove this mirror on next upgrade, it's only needed because upstream took ncurses-6.0-20171125.tgz down!
-      "http://bld1.alpinelinux.org/distfiles/v3.5/ncurses-${version}.tgz"
-
-      "ftp://ftp.invisible-island.net/ncurses/current/ncurses-${version}.tgz"
-      "https://invisible-mirror.net/archives/ncurses/current/ncurses-${version}.tgz"
-    ];
-    sha256 = "11adzj0k82nlgpfrflabvqn2m7fmhp2y6pd7ivmapynxqb9vvb92";
+    url = "mirror://gnu/ncurses/${name}.tar.gz";
+    sha256 = "05qdmbmrrn88ii9f66rkcmcyzp1kb1ymkx7g040lfkd1nkp7w1da";
   };
 
   patches = lib.optional (!stdenv.cc.isClang) ./clang.patch;

